trestle/core/rules_validator.py

```diff
@@ -0,0 +1,110 @@
+# -*- mode:python; coding:utf-8 -*-
+
+# Copyright (c) 2023 IBM Corp. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Validate by confirming rule parameter values are consistent."""
+import logging
+import pathlib
+
+import trestle.oscal.catalog as cat
+from trestle.common.common_types import TopLevelOscalModel
+from trestle.common.common_types import TypeWithProps
+from trestle.common.list_utils import as_list, deep_get, deep_set, get_item_from_list
+from trestle.core.catalog.catalog_interface import CatalogInterface
+from trestle.core.profile_resolver import ProfileResolver
+from trestle.core.validator import Validator
+from trestle.oscal.ssp import SystemSecurityPlan
+
+logger = logging.getLogger(__name__)
+
+shared_params = {}
+
```

Comment:
Please avoid using globals unless there is a very good reason you want to share it throughout the code - which is very rare.  Also - I would call this `rule_param_dict` as it is only called shared params in the particular data set.  The intent here is to gather all rule_params and associated values - and then check for any that have multiple different values across controls.  It is also a dict rather than a simple list, so I would make that clear.

Reply:
changed.